IQSS/dataverse-frontend
# 344 - Display real collection data in the Collection Page

## What this PR does / why we need it:
This PR adds the integration with js-dataverse getCollection use case to display real collection data in the Collection Page.

Depends on:
- https://github.com/IQSS/dataverse-client-javascript/issues/131

## Which issue(s) this PR closes:

- Closes #344 

## Special notes for your reviewer:
e2e MetadataBlock test is failing because of some recent changes to the API. This is fixed in another PR:

- #334 

The UI does not exactly match that of JSF, as best practices dictate that breadcrumbs should always be positioned at the top, before anything else

![image](https://github.com/IQSS/dataverse-frontend/assets/23359572/bdea2ec4-9731-4221-a458-96d2e084886d)

## Suggestions on how to test this:
### Step 1: Run the Development Environment
1. Execute `npm i`.
2. Navigate with `cd packages/design-system && npm run build`.
3. Return with `cd ../../`.
4. Ensure you have a `.env` file similar to `.env.example`, with the variable `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000`.
5. Navigate with `cd dev-env`.
6. Start the environment using `./run-env.sh unstable`.
7. To verify the environment, visit <http://localhost:8000> and check your local Dataverse installation.

### Step 2: Test the Form
1. Navigate to <http://localhost:8000>.
2. Log in as an admin using the username: `dataverseAdmin` and password: `admin1`.
3. Create a new Dataverse collection.
4. Go to <http://localhost:8000/spa/collections?id=<your-new-collection-alias>>
5. Check that the collection data is correct and that the breadcrumbs are correct.

## Does This PR Introduce a User Interface Change? If Mockups Are Available, Please Link/Include Them Here:
![image](https://github.com/IQSS/dataverse-frontend/assets/23359572/01703c0e-621d-499d-bb29-1a857d979278)

## Is There a Release Notes Update Needed for This Change?:
Collection Page now displays real collection data
